core/src/rpc.rs

```diff
@@ -382,6 +382,60 @@ impl JsonRpcRequestProcessor {
         Ok(result)
     }
 
+    pub fn get_inflation_reward(
+        &self,
+        pubkey: &Pubkey,
+        epoch: Epoch,
+    ) -> Result<Option<RpcInflationReward>> {
+        let epoch_schedule = self.get_epoch_schedule();
+        let first_available_block = self.get_first_available_block();
+
+        // Rewards for this epoch are found in the first confirmed block of the next epoch
+        let first_slot_in_epoch = epoch_schedule.get_first_slot_in_epoch(epoch + 1);
+        if first_slot_in_epoch < first_available_block {
+            return Err(Error::internal_error());
```

Comment:
This should probably be RpcCustomError::BlockNotAvailable?

Reply:
@CriesofCarrots any idea what error would make the most sense here?

Reply:
It seems like the correct errors are either `LongTermStorageSlotSkipped`, if bigtable was queried by `first_available_block()`, or `SlotCleanedUp` if only blockstore. You'll probably have to use:
```
if self.bigtable_ledger_storage.is_some()
```